lib/Dialect/SV/Transforms/GeneratorCallout.cpp

```diff
@@ -0,0 +1,159 @@
+    genOptions.push_back(temp.substr(pos + 1));
+    temp = temp.substr(0, pos);
+  }
+  genOptions.push_back(temp);
+  temp = genExec;
+
+  auto pathLoc = temp.find_last_of("/\\");
+  if (pathLoc != std::string::npos)
+    genExecutablePath = temp.substr(0, pathLoc + 1);
+  else
+    genExecutablePath = "";
+  execName = temp.substr(pathLoc + 1);
+  return;
+}
+
+void RTLGeneratorCalloutPass::runOnOperation() {
+  ModuleOp root = getOperation();
+  std::string genExecutableName, genExecutablePath;
+  std::vector<std::string> genOptions;
+  parseProgramArgs(genExecutable, genExecArgs, genExecutableName,
+                   genExecutablePath, genOptions);
+
+  OpBuilder builder(root->getContext());
+  auto genMemExe =
+      llvm::sys::findProgramByName(genExecutableName, {genExecutablePath});
+  // If the executable was not found in the provided path (when the path is
+  // empty), then search it in the $PATH.
+  if (!genMemExe)
+    genMemExe = llvm::sys::findProgramByName(genExecutableName);
+  // If cannot find the executable, then nothing to do return.
+  if (!genMemExe)
+    return;
+  SmallVector<Operation *> opsToRemove;
+  for (auto &op : *root.getBody()) {
+    if (auto modGenOp = dyn_cast<RTLModuleGeneratedOp>(op)) {
+      // Get the corresponding schema associated with this generated op.
+      if (auto genSchema =
+              dyn_cast<RTLGeneratorSchemaOp>(modGenOp.getGeneratorKindOp())) {
+        if (genSchema.descriptor().str() != schemaName)
+          continue;
+        std::vector<std::string> generatorArgs;
+        // First argument should be the executable name.
+        generatorArgs.push_back(genExecutableName);
+
+        for (auto o : genOptions) {
+          generatorArgs.push_back(o);
+        }
+        auto moduleName = modGenOp.getVerilogModuleNameAttr().getValue().str();
+        // The moduleName option is not present in the schema, so add it
+        // explicitly.
+        generatorArgs.push_back("--moduleName");
+        generatorArgs.push_back(moduleName);
+        // Iterate over all the attributes in the schema.
+        // Assumption: All the options required by the generator program must be
+        // present in the schema.
+        for (auto attr : genSchema.requiredAttrs()) {
+          // Get the port name from schema.
+          StringRef portName = attr.dyn_cast<StringAttr>().getValue();
+          generatorArgs.push_back("--" + portName.str());
+          // Get the value for the corresponding port name.
+          // Assumption: the value always exists.
+          auto v = modGenOp->getAttr(portName);
+          if (auto intV = v.dyn_cast<IntegerAttr>())
+            generatorArgs.push_back(intV.getValue().toString(10, false));
+          else if (auto strV = v.dyn_cast<StringAttr>())
+            generatorArgs.push_back(strV.getValue().str());
+        }
+        generatorArgs.push_back(" ");
+        std::vector<StringRef> generatorArgStrRef;
```

Comment:
SmallVector.  Also, you should be able to construct this earlier.

Reply:
Still need a separate loop to create `StringRef` vector, can't use the temporary string inside the loop.